### ISSUE TITLE
cmd/vet: no or incorrect warning for zero-based indexing in printf format

### ISSUE THREAD

**Issue by jstemmer** [golang/go#9752]
Using go1.4.1 linux/amd64 and golang.org/x/tools `1afe783e9f23b46389ab7a439ed4e0eee558382a`.

Run the following source ([playground link](http://play.golang.org/p/9K7Z4qNqDa)) and then run `go vet` on the same file:
```go
fmt.Printf("%[0]s\n", "arg1")
fmt.Printf("%[0]d\n", 1)
```

You'll see the following output:
```
$ go run vet.go
%!s(BADINDEX)
%!d(BADINDEX)

$ go vet vet.go
vet.go:9: arg "%[0]d\n" for printf verb %d of wrong type: string
exit status 1
```

There was no warning for the first format and a misleading warning for the second format.

`vet` does warn for negative or out of bounds indices, so for consistency I think it should also warn that indices are one-based.

**Comment by jstemmer** [comment]
Proposed fix: https://go-review.googlesource.com/#/c/3692/

**Comment by jstemmer** [comment]
Fixed in golang/tools@913f41fc5f1b7afd8f7c66c011c1ad368fa23fb8
